lib/components/CdeModal.tsx

```diff
@@ -0,0 +1,65 @@
+import {FC, useEffect} from 'react';
```

Comment:
This is the base Modal of our library, I think we can have just this one for the whole application pipeline and we change the content of the modal based on the step. As first iteration I added a switch to address this goal.